### PR TITLE
CM-848: Setup CodeQL

### PR DESCRIPTION
**Pull request by wi101**
[CM-848](https://liveintent.atlassian.net/browse/CM-848)

Remove lgtm and setup CodeQL

Author Todo List:

- [x] Add/adjust tests (if applicable)
- [x] Build in CI passes
- [x] Latest master revision is merged into the branch
- [x] Self-Review
- [x] Set `Ready For Review` status

[CM-848]: https://liveintent.atlassian.net/browse/CM-848?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/codeql/codeql-config.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+paths:
+  - src
+  - rollup
```
[the remark this answers is not in the document]
Comment (wi101) [reply .github/codeql/codeql-config.yml:3]: I saw there are some vulnerabilities
https://github.com/LiveIntent/live-connect/security/code-scanning
and then I thought it should be ok to not scan tests. but I can put it back and then we can fix the issues

[the remark this answers is not in the document]
Comment (SuperIzya) [reply .github/codeql/codeql-config.yml:3]: Can we put a comment in code for CodeQL to skip next line? I would prefer to be aware there is smth fishy with tests...

File: README.md [hunk 2]
```diff
@@ -3,8 +3,7 @@
...
 [![CircleCI](https://circleci.com/gh/LiveIntent/live-connect/tree/master.svg?style=svg)](https://circleci.com/gh/LiveIntent/live-connect/tree/master)
 [![codecov](https://codecov.io/gh/liveintent-berlin/live-connect/branch/master/graph/badge.svg?token=P5sRpM4U6k)](https://codecov.io/gh/liveintent-berlin/live-connect)
-[![Total alerts](https://img.shields.io/lgtm/alerts/g/LiveIntent/live-connect.svg?logo=lgtm&logoWidth=18)](https://lgtm.com/projects/g/LiveIntent/live-connect/alerts/)
-[![Language grade: JavaScript](https://img.shields.io/lgtm/grade/javascript/g/LiveIntent/live-connect.svg?logo=lgtm&logoWidth=18)](https://lgtm.com/projects/g/LiveIntent/live-connect/context:javascript)
+[![CodeQL](https://github.com/LiveIntent/live-connect/actions/workflows/codeql.yml/badge.svg?branch=master&event=push)](https://github.com/LiveIntent/live-connect/actions/workflows/github-code-scanning/codeql)
```
Comment (3link) [inline README.md:6]: What is the best way to check if it works?

Comment (wi101) [reply README.md:6]: it is when we use the link with this branch name

Comment (wi101) [reply README.md:6]: [CodeQL](https://github.com/LiveIntent/live-connect/actions/workflows/github-code-scanning/codeql)

Comment (3link) [reply README.md:6]: Thanks, that works. Used that link: https://github.com/LiveIntent/live-connect/actions/workflows/codeql.yml/badge.svg?branch=no-ticket_setup-codeql&event=push

**Review by 3link — APPROVED** [review]

File: .github/codeql/codeql-config.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (wi101) [reply .github/codeql/codeql-config.yml:4]: I couldn't find a way how to do it, but I will put the test back and we can dismiss the alert as it is used in Test in that case.. At least we will be aware of that.

Screenshot 2023-06-14 at 09 33 30

**Review by SuperIzya — APPROVED** [review]

**Pull request merged**